### PR TITLE
Fetch Messages by Date (instead of UID) when going through Folder's history

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1971

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/models/Folder.kt [hunk 1]
```diff
@@ -71,10 +70,17 @@ class Folder : RealmObject {
...
     var messages: RealmList<Message> = realmListOf()
+
+    /**
+     * List of old Messages UIDs of this Folder that we need to fetch.
+     * When first opening the Folder, we get the full list of UIDs, and we store it.
+     * Then, we'll be able to go through it as we want to fetch the old Messages.
+     */
     @Transient
-    var remainingOldMessagesToFetch: Int = DEFAULT_REMAINING_OLD_MESSAGES_TO_FETCH
+    var oldMessagesUidsToFetch = realmListOf<Int>()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/models/Folder.kt:80]: Check if we receive 50 new Threads when fetching Messages.
If not, continue looping deeper into `oldMessagesUidsToFetch` until 50 new Threads have been added in the Folder.

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt [hunk 2]
```diff
@@ -314,205 +208,143 @@ class RefreshController @Inject constructor(
...
+    private suspend fun Realm.fetchOnePageOfOldMessages(scope: CoroutineScope, folderId: String) {
 
-    private suspend fun Realm.fetchAllOldPages(scope: CoroutineScope, folder: Folder) {
+        var totalNewThreads = 0
+        var upToDateFolder = getUpToDateFolder(folderId)
+        var maxPagesToFetch = Utils.MAX_OLD_PAGES_TO_FETCH_TO_GET_ENOUGH_NEW_THREADS
 
-        fun remainingCount() = FolderController.getFolder(folder.id, realm = this)?.remainingOldMessagesToFetch ?: -1
+        while (
+            totalNewThreads < Utils.PAGE_SIZE / 2 &&
```
Comment (PhilippeWeidmann) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt:233]: What happens if `fetchOnePage` returns 0 ?

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt:233]: Discussed in DM.

File: app/src/main/java/com/infomaniak/mail/data/models/Folder.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/models/Folder.kt:81]: Done.

The limit has been fixed at `PAGE_SIZE / 2` _(25)_.

**Review by tevincent — APPROVED** [review]

**Pull request merged**